sklearn/linear_model/tests/test_base.py

```diff
@@ -93,6 +93,55 @@ def test_linear_regression_sample_weights():
                 assert_almost_equal(inter1, coefs2[0])
 
 
+def test_sample_weights():
```

Comment:
Part of this is already covered by `check_sample_weights_invariance` in common tests (see `sklearn/utils/estimator_checks.py`) and I would rather we added more checks there, rather than add estimator specific tests that would take effort to maintain in the long run.